rewrite-pattern.js

```diff
@@ -166,10 +171,12 @@ const processCharacterClass = (characterClassItem, regenerateOptions) => {
 				throw new Error(`Unknown term type: ${ item.type }`);
 		}
 	}
-	if (characterClassItem.negative) {
-		update(characterClassItem, `(?!${set.toString(regenerateOptions)})[\\s\\S]`)
-	} else {
-		update(characterClassItem, set.toString(regenerateOptions));
+	if (transformed) {
```

Comment:
I added this check so that when not transforming the `u` flag, things like `/[\uD806\uDCDF]/u` are not modified.